### PR TITLE
feat: made product variants immutable

### LINKED ISSUE #293: Implement immutable inventory for products

### PR DESCRIPTION
**Pull request by callmekungfu**
### Related Issue(s)

resolves #293 

### Quick overview of the change
CleanShot 2022-10-19 at 15 39 44@2x

https://user-images.githubusercontent.com/11084692/196820188-0e6a490b-51d2-48c3-9f14-ed3c1c8b20a9.mp4

- UI changes in activity entry
- Updated UI flow for updating variants

### CODE REVIEW

**Comment by callmekungfu** [comment]
Blocked by #292, once merged we will move to review

File: client/src/pages/inventory/VariantCreate.tsx
Comment (andrewpratheepan) [inline client/src/pages/inventory/VariantCreate.tsx:148]: Can we type this now? I can see that the variant can either be `Variant` or `PostVariantBody`, the latter of which does not have id. Just wanted to ask how we could tackle this.

**Review by andrewpratheepan — APPROVED** [review]

File: client/src/pages/inventory/VariantCreate.tsx
Comment (callmekungfu) [reply client/src/pages/inventory/VariantCreate.tsx:148]: We need to look at why there is a delta in the component, why is one component handling the duty of two vastly different flows. This requires a larger scoped rework than we might anticipate so I say this is something we should dedicate some time to in the future.

The page was not well implemented before actually, it didn't even have variant PUT integrated, so whenever you update a variant it just POSTs a new one, I just got it to work for now but if we ever get time and evolve our product we should tackle it.

**Pull request merged**
